src/main/kotlin/fi/metatavu/oss/api/impl/realtime/mqtt/AbstractReactiveProducer.kt

```diff
@@ -0,0 +1,65 @@
+package fi.metatavu.oss.api.impl.realtime.mqtt
+
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import io.smallrye.reactive.messaging.mqtt.MqttMessage
+import org.eclipse.microprofile.config.inject.ConfigProperty
+import org.eclipse.microprofile.reactive.messaging.Emitter
+import java.util.*
+import javax.inject.Inject
+
+/**
+ * Abstract base class for reactive MQTT message producers
+ */
+abstract class AbstractReactiveProducer<T> {
```

Comment:
This contains device specific methods, so should we either:

rename this to e.g. AbstractDeviceReactiveProducer or keep this and create another device specific class that extends this class?

Reply:
Resolved